implementations/python/sugar-lift-py-tests/tests/test_small_family_batch.py

```
@@ -0,0 +1,161 @@
+    direct = named[0].desugar(None)
+    assert isinstance(direct, Complete)
+    assert direct.value.name == "n"  # type: ignore[attr-defined]
+
+
+def test_named_expr_non_name_target_stays_loud(tmp_path: Path) -> None:
+    # Invalid in CPython for attribute walrus in some positions; use a
+    # structural check via direct construction if parse allows.
+    # Skip if parser rejects — the loud arm is the contract.
+    try:
+        sf = _file(tmp_path, "def f(o, z):\n    (o.x := z)\n    return o\n")
+    except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)